pkg/controller/status/status.go

```diff
@@ -1031,3 +1033,12 @@ func UpdateStatusCondition(statuscondition []metav1.Condition, conditions []oper
 	}
 	return statuscondition
 }
+
+func SetDegraded(status StatusManager, reason operator.TigeraStatusReason, message string, err error, log logr.Logger) {
```

Comment:
There are two issues/questions I currently have with status.go: 
- Not this func, but the method SetDegraded(string, string) should no longer accept a string for the reason, but only operator.TigeraStatusReason. Otherwise, new errors may creep in over time.
-  Do we still need the existing method (so not this func)? If we do, can it perhaps be unexported?

Reply:
Updated the syntax of SetDegraded(string, string)  to SetDegraded(operator.TigeraStatusReason, string) 
Lock and unlock of the status manager instance happens in existing SetDegraded func, if we need to remove it.. then we need to update the current method `func SetDegraded(status StatusManager,..)` as receiver function and perform the lock and unlock here
```
func (m *statusManager) SetDegraded(status StatusManager, reason operator.TigeraStatusReason, message string, err error, log logr.Logger) {
	log.WithValues(string(reason), message).Error(err, string(reason))
	errormsg := ""
	if err != nil {
		errormsg = err.Error()
	}
	m.lock.Lock()
	defer m.lock.Unlock()
	m.degraded = true
	m.explicitDegradedReason = string(reason)
	m.explicitDegradedMsg = fmt.Sprintf("%s - Error: %s", message, errormsg)
}
```

Reply:
I guess this is enough then:

>  1)Updated the syntax of SetDegraded(string, string) to SetDegraded(operator.TigeraStatusReason, string)

And leave the other bullet unaddressed.